study.md

```diff
@@ -21,18 +21,18 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+<!-- I was confused about hoisting.  How do the variables know to move up before the alert or return statement. -->
 ```
 
 Are there any nagging questions relating to functions that you have?
 
 ```md
-<!-- your answer here -->
+<!-- How does the call and apply function replace the this object? -->
```

Comment:
```call``` and ```apply``` are used to invoke a method and set its reference of ```this``` to the first argument.  The second argument to ```call``` is the argument to be passed into the method being invoked.  ```apply``` is similar, but takes an array of arguments.  Here's an example that I wrote.  Let me know if it's unclear.
```javascript
function Person(name) {
  this.name = name;
}

Person.prototype.greet = function(greeting) {
  console.log(greeting);
  console.log('This is:', this);
};

Person.prototype.greetThreeTimes = function(first, second, third) {
  console.log(first);
  console.log(second);
  console.log(third);
  console.log('This is:', this);
};

const hank = new Person('Hank');
const peggy = new Person('Peggy');
const bobby = new Person('Bobby');

hank.greet('Hello.');
// Output:
// Hello.
// This is: Person { name: 'Hank' }

hank.greet.call(peggy, 'Hi.');
// Output:
// Hi.
// This is: Person { name: 'Peggy' }

hank.greetThreeTimes.apply(bobby, ['Hello.', 'Hi.', 'Greetings']);
// Output:
// Hello.
// Hi.
// Greetings
// This is: Person { name: 'Bobby' }
```